src/test/java/com/stripe/functional/BankAccountTest.java

```diff
@@ -0,0 +1,166 @@
+
+    return bankAccount;
+  }
+
+  @Test
+  public void testCreate() throws StripeException {
+    final Customer customer = Customer.retrieve(CUSTOMER_ID);
+
+    Map<String, Object> params = new HashMap<String, Object>();
+    params.put("source", "btok_123");
+
+    final BankAccount bankAccount = (BankAccount) customer.getSources().create(params);
+
+    assertNotNull(bankAccount);
+    verifyRequest(
+        APIResource.RequestMethod.POST,
+        String.format("/v1/customers/%s/sources", customer.getId()),
+        params
+    );
+  }
+
+  @Test
+  public void testRetrieve() throws StripeException {
+    final Customer customer = Customer.retrieve(CUSTOMER_ID);
+
+    final BankAccount bankAccount = (BankAccount) customer.getSources().retrieve(BANK_ACCOUNT_ID);
+
+    assertNotNull(bankAccount);
+    verifyRequest(
+        APIResource.RequestMethod.GET,
+        String.format("/v1/customers/%s/sources/%s", customer.getId(), BANK_ACCOUNT_ID)
+    );
+  }
+
+  @Test
+  public void testUpdate() throws IOException, StripeException {
+    final Customer customer = Customer.retrieve(CUSTOMER_ID);
+    final BankAccount bankAccount = getBankAccountFixture(customer);
+
+    Map<String, Object> metadata = new HashMap<String, Object>();
+    metadata.put("key", "value");
+    Map<String, Object> params = new HashMap<String, Object>();
+    params.put("metadata", metadata);
+
+    // stripe-mock returns a Card instance instead of a BankAccount
+    stubRequest(
+        APIResource.RequestMethod.POST,
+        String.format("/v1/customers/%s/sources/%s", customer.getId(), bankAccount.getId()),
+        params,
+        BankAccount.class,
+        getResourceAsString("/api_fixtures/bank_account.json")
+    );
+
+    final BankAccount updatedBankAccount = bankAccount.update(params);
+
+    assertNotNull(updatedBankAccount);
+    verifyRequest(
+        APIResource.RequestMethod.POST,
+        String.format("/v1/customers/%s/sources/%s", customer.getId(), bankAccount.getId()),
+        params
+    );
+  }
+
+  @Test
+  public void testList() throws IOException, StripeException {
+    final Customer customer = Customer.retrieve(CUSTOMER_ID);
+
+    Map<String, Object> params = new HashMap<String, Object>();
+    params.put("object", "bank_account");
+
+    // stripe-mock doesn't handle this, so we stub the request
+    BankAccount stubbedBankAccount = APIResource.GSON.fromJson(
+        getResourceAsString("/api_fixtures/bank_account.json"), BankAccount.class);
+    ExternalAccountCollection stubbedCollection = new ExternalAccountCollection();
+    List<ExternalAccount> stubbedData = new LinkedList<ExternalAccount>();
+    stubbedData.add(stubbedBankAccount);
+    stubbedCollection.setData(stubbedData);
+    stubRequest(
+        APIResource.RequestMethod.GET,
+        String.format("/v1/customers/%s/sources", customer.getId()),
+        params,
+        ExternalAccountCollection.class,
+        stubbedCollection.toJson()
```

Comment:
I don't really understand this one. Why not a fixture that is a list already? I think that's what I did for other tests.

Reply:
Because I'd be mildly annoyed to have a fixture file with a bank account, and a different fixture file with the same bank account wrapped in a list.

Hopefully stripe-mock will be able to handle this case sooner rather than later and those fixtures will become unnecessary anyway.

Reply:
FWIW I did that for other resources though.